state-chain/pallets/cf-pools/src/lib.rs

```diff
@@ -837,12 +931,17 @@ pub mod pallet {
 		#[pallet::weight(T::WeightInfo::schedule_limit_order_update())]
 		pub fn schedule_limit_order_update(
 			origin: OriginFor<T>,
-			call: Box<Call<T>>,
-			dispatch_at: BlockNumberFor<T>,
-		) -> DispatchResultWithPostInfo {
+			base_asset: any::Asset,
+			quote_asset: any::Asset,
+			side: Side,
+			id: OrderId,
+			details: LimitOrderUpdateDetails<BlockNumberFor<T>>,
+			dispatch_at: Option<BlockNumberFor<T>>,
+		) -> DispatchResult {
 			let lp = T::AccountRoleRegistry::ensure_liquidity_provider(origin)?;
 
-			Self::schedule_or_dispatch_limit_order_update(lp, call, dispatch_at)
+			LimitOrderUpdate::<T> { lp: lp.clone(), base_asset, quote_asset, side, id, details }
+				.schedule_or_dispatch(dispatch_at)
```

Comment:
If this is the only place where `schedule_or_dispatch` is called I think it might make sense to inline it? (If we had more code in `schedule_limit_order_update`, it would probably be justified to split it up, but now it looks like there isn't much benefit to having a separate function).

Reply:
Yes I think I agree.